docs/features/tuples.md

```diff
@@ -230,3 +230,65 @@ M1((1, ()=>(2, 3)));         // the first overload is used due to "exact match"
 
 ``` 
 
+Language grammar changes
+---------------------
+This is based on the [ANTLR grammar](https://raw.githubusercontent.com/ljw1004/csharpspec/gh-pages/csharp.g4) from Lucian.
+
+For tuple type declarations:
+
+```ANTLR
+struct_type
+    : type_name
+    | simple_type
+    | nullable_type
+    | tuple_type // new
+    ; 
+    
+tuple_type
+    : '(' tuple_type_element_list ')'
+    ;
+    
+tuple_type_element_list
+    : tuple_type_element ',' tuple_type_element
+    | tuple_type_element_list ',' tuple_type_element
+    ;
+    
+tuple_type_element
+    : type_parameter identifier?
+    ;
+```
+
+For tuple literals:
+
+```ANTLR
+literal
+    : boolean_literal
+    | integer_literal
+    | real_literal
+    | character_literal
+    | string_literal
+    | null_literal
+    | tuple_literal // new
+    ;
+
+tuple_literal
+    : '(' tuple_literal_element_list ')'
+    ;
+
+tuple_literal_element_list
+    : tuple_literal_element ',' tuple_literal_element
+    | tuple_literal_element_list ',' tuple_literal_element
+    ;
+
+tuple_literal_element
+    : ( identifier ':' )? literal
+    ;
+```
+
+A tuple literal cannot be used as default value for an optional parameter.
```

Comment:
This is non-normative (it is implied by rules elsewhere), so perhaps precede by "Because it is not a constant expression, "